poly-synth/src/poly-synth-demo.js

```diff
@@ -22,163 +22,368 @@ class PolySynthDemo {
           name: 'Sustain',
           id: 'filterSustain',
           type: 'range',
           min: this.polySynth_.minSustain,
           max: this.polySynth_.maxSustain,
           step: 0.01,
-          default: this.polySynth_.getParameters().filterSustain,
+          default: this.polySynth_.getParameters().filterSustain
         });
 
     let filterReleaseSlider_ = new ParamController(
-        filterADSRId, this.polySynth_.setParameter.bind(this.polySynth_), {
+        containerId, this.polySynth_.setParameter.bind(this.polySynth_), {
           name: 'Release (s)',
           id: 'filterRelease',
           type: 'range',
           min: this.polySynth_.minRelease,
           max: this.polySynth_.maxRelease,
           step: 0.01,
-          default: this.polySynth_.getParameters().filterRelease,
+          default: this.polySynth_.getParameters().filterRelease
         });
 
     let filterDetuneSlider_ = new ParamController(
-        filterADSRId, this.polySynth_.setParameter.bind(this.polySynth_), {
+        containerId, this.polySynth_.setParameter.bind(this.polySynth_), {
           name: 'Detune Amount',
           id: 'filterDetuneAmount',
           type: 'range',
           min: this.polySynth_.minDetuneAmount,
           max: this.polySynth_.maxDetuneAmount,
           step: 0.01,
-          default: this.polySynth_.getParameters().filterDetuneAmount,
+          default: this.polySynth_.getParameters().filterDetuneAmount
+        });
+  }
+  
+  initializeBitcrusherGUI_(containerId){
+    let bitcrusherBitDepthSlider_ = new ParamController(
+        containerId, this.polySynth_.setBitDepth.bind(this.polySynth_), {
+          name: 'Bit Depth',
+          id: 'bitDepth',
+          type: 'range',
+          min: this.polySynth_.minBitDepth,
+          max: this.polySynth_.maxBitDepth,
+          step: 0.01,
+          default: this.polySynth_.bitcrusher.bitDepth
         });
+    bitcrusherBitDepthSlider_.disable();
+
+    let bitcrusherReductionSlider_ = new ParamController(
+        containerId, this.polySynth_.setReduction.bind(this.polySynth_), {
+          name: 'Reduction',
+          id: 'reduction',
+          type: 'range',
+          min: this.polySynth_.minReduction,
+          max: this.polySynth_.maxReduction,
+          step: 1,
+          default: this.polySynth_.bitcrusher.reduction
+        });
+    bitcrusherReductionSlider_.disable();
+
+    // Sound is not processed by the bitcrusher if in bypass mode.
+    document.getElementById('bitcrusherBypassButton').onclick = (event) => {
+      // Only one of |this.polySynth_.activeBitcrusherRoute_| and
+      // |this.polySynth_.bypassBitcrusherRoute_| has a non-zero gain.
+      if (event.target.textContent === 'Active') {
+        event.target.textContent = 'Bypassed';
+
+        // The change is scheduled slightly into the future to avoid glitching.
+        const t = this.context_.currentTime + 0.01;
```

Comment:
You certainly hear that a swapping is occurring but its not a pronounced glitch as it is if the swap occurs immediately